### PR TITLE
added lu factor to torch frontend and tested

### LINKED ISSUE #18377: lu

### PR DESCRIPTION
**Pull request by Rakshith-611**
Close #18377

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py [hunk 1]
```diff
@@ -1355,3 +1355,29 @@ def test_torch_solve_ex(
         B=other,
         check_errors=check,
     )
+
+@handle_frontend_test(
+    fn_tree="torch.linalg.lu_factor", input_dtype_and_input=_lu_factor_helper()
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py:1360]: `_lu_factor_helper()` is not defined

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hello! Thank you for your contribution! Can you please resolve the issue?

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hey! Your implemenation is failing due to following error now:
```
E       NameError: name 'input_dtype' is not defined
```

**Comment by NripeshN** [comment]
The code

**Pull request closed without merging**